Assets/Scripts/ScrapCollector.cs

```diff
@@ -0,0 +1,92 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ScrapCollector : MonoBehaviour
+{
+    [SerializeField] private ScrapManager scrapManager;
+    [SerializeField] private List<PlayerController> enemyMecha;
+    [SerializeField] private List<PlayerController> highHealthMecha;
+    private int scrapAvailable;
+    
+    // Start is called before the first frame update
+    void Start()
+    {
+        CollectEnemyMecha();
+    }
+
+    private void Update()
+    {
+        scrapAvailable = scrapManager.GetScrapAvailable();
```

Comment:
Sounds good. I'll make it an argument of the AddScrap and SubtractScrap methods. I wanted to make sure that whenever AddScrap or SubtractScrap is called, we have the most updated version of scrapAvailable, but since we're not really calling either method much, I guess it's better to just have it as an argument.